### PR TITLE
Fix #173: Use OpenShift compliant git/vcs annotations

### LINKED ISSUE #173: Use OpenShift compliant git/vcs annotations
## Description
As of now, Elipse JKube adds the following annotations to every generated resource manifest `jkube.io/git-branch`, `jkube.io/git-url`, `jkube.io/git-commit`.

We should include ~or replace~ the ones recommended in @gorkem [app-label guidelines](https://github.com/gorkem/app-labels/blob/master/labels-annotation-for-openshift.adoc)(`app.openshift.io/vcs-uri`, `app.openshift.io/vcs-ref`).

https://github.com/eclipse/jkube/blob/b294c87433532762234a61e95b3d939749b4aff8/jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/GitEnricher.java#L57-L81

## See also
- https://groups.google.com/d/msgid/fabric8-devclients/e1dcc1b7-ac71-4761-8917-0d6835b6b32f%40googlegroups.com?utm_medium=email&utm_source=footer

### PR DESCRIPTION
**Pull request by rohanKanojia**
Fix #173

### CODE REVIEW

File: jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/GitEnricher.java [hunk 1]
```diff
@@ -101,58 +81,80 @@ public void create(PlatformMode platformMode, KubernetesListBuilder builder) {
...
         builder.accept(new TypedVisitor<JobBuilder>() {
             @Override
             public void visit(JobBuilder builder) {
-                builder.editMetadata().addToAnnotations(getAnnotations()).endMetadata();
+                builder.editMetadata().addToAnnotations(addAnnotations(platformMode)).endMetadata();
             }
         });
     }
+
+    protected void addAnnotations(PlatformMode platformMode, Map<String, String> annotations, String gitRemoteUrl, String branch, String commitId) {
```
Comment (manusa) [inline jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/GitEnricher.java:138]: It would be better if the function was pure and instead of creating side effects in the input annotations variable you returned a new Map with the annotations. This map can then be appended (if applicable) to the existing variable using `putAll`.

File: jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/GitEnricher.java [hunk 2]
```diff
@@ -101,58 +81,84 @@ public void create(PlatformMode platformMode, KubernetesListBuilder builder) {
...
         builder.accept(new TypedVisitor<JobBuilder>() {
             @Override
             public void visit(JobBuilder builder) {
-                builder.editMetadata().addToAnnotations(getAnnotations()).endMetadata();
+                builder.editMetadata().addToAnnotations(addAnnotations(platformMode)).endMetadata();
             }
         });
     }
+
+    protected static Map<String, String> addAnnotations(PlatformMode platformMode, String gitRemoteUrl, String branch, String commitId) {
```
Comment (manusa) [inline jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/GitEnricher.java:138]: nit: this method is not adding any annotations but generating/inferring them. Method should be named something like `getAnnotations` `generateAnnotations` `inferAnnotations` or sth like that.

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/GitEnricherTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import org.eclipse.jkube.kit.config.resource.JKubeAnnotations;
+import org.eclipse.jkube.kit.config.resource.OpenShiftAnnotations;
+import org.eclipse.jkube.kit.config.resource.PlatformMode;
+import org.junit.Test;
+
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+public class GitEnricherTest {
```
Comment (manusa) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/GitEnricherTest.java:25]: Some corner or edge cases should be tested: null method parameter values, etc.
